tests/vfolders/path_matcher.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+import pytest
+
+from pootle_fs.utils import PathFilter
+from pootle_store.models import Store
+from virtualfolder.models import VirtualFolder
+from virtualfolder.utils import VirtualFolderPathMatcher
+
+
+@pytest.mark.pootle_vfolders
+@pytest.mark.django_db
+def test_vfolder_path_matcher():
+    return
+    vfolder = VirtualFolder.objects.create(
+        name="avfolder",
+        filter_rules="FOO,BAR")
+    path_matcher = VirtualFolderPathMatcher(vfolder)
+    assert path_matcher.vf == vfolder
+    assert path_matcher.tp_path == "/[^/]*/[^/]*/"
+    assert isinstance(
+        vfolder.path_matcher,
+        VirtualFolderPathMatcher)
+    # no projects/languages set by default
+    assert (
+        list(vfolder.path_matcher.languages)
+        == list(vfolder.path_matcher.projects)
+        == list(vfolder.path_matcher.existing_stores)
+        == [])
+
+
+@pytest.mark.pootle_vfolders
+@pytest.mark.django_db
+def test_vfolder_path_matcher_languages(language0, language1):
+    vfolder = VirtualFolder.objects.create(
+        name="avfolder",
+        filter_rules="FOO,BAR")
+    vfolder.languages.add(language0)
+    vfolder.languages.add(language1)
+    # no projects set so no stores
+    assert list(vfolder.path_matcher.existing_stores) == []
+    assert (
+        sorted(vfolder.path_matcher.languages)
+        == [language0.pk, language1.pk])
+    vfolder.all_languages = True
+    vfolder.save()
+    assert vfolder.path_matcher.languages is None
+    assert list(vfolder.path_matcher.projects) == []
```

Comment:
This should match stores in all TPs. But without TPs and stores it can't retrieve anything.